moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1725,6 +1725,9 @@ as the new values that correspond to the group */
       const int fvi = mim[i]->getFirstVariableIndex();
       position_[fvi] =
           mim[i]->getMimicFactor() * position_[mim[i]->getMimic()->getFirstVariableIndex()] + mim[i]->getMimicOffset();
+      // Only mark joint transform dirty, but not the associated link transform
+      // as this function is always used in combination of
+      // updateMimicJoint(group->getMimicJointModels()) + markDirtyJointTransforms(group);
```

Comment:
this "always used in combination of" rule should be in the function comment header, line 1720

Reply:
To avoid this sloppy rule at all, I suggested to merge both functions into `updateMimicJoints(group)` as implemented in https://github.com/ubi-agni/moveit/commit/c95b033e221a434bc16828d4af2efdf220ed99f3. @davetcoleman What do you think about that? If you like the idea, I will augment to this PR. How should we handle API breakage in this case? Do you think it's necessary to deprecate the old functions?

Reply:
These changes feel dangerous - could you write some unit tests for ``RobotState`` that ensure behavior is maintained? Then I would be comfortable with the implementation by @ubi-agni 

I am a fan of using MoveIt!'s function deprecation macro for the old functions, yes

Reply:
No, I don't have time to write a unittest for this in the near future. Furthermore, I don't see which kind of unittest could validate my changes. The function pair `updateMimicJoint(group->getMimicJointModels())` and `markDirtyJointTransforms(group)` is used all over and I merged all occurences into `updateMimicJoints(group)`. 
A unittest to verify that the old pair of functions and the proposed new function do the same job is trivial on the other hand.

Reply:
> A unittest to verify that the old pair of functions and the proposed new function do the same job is trivial on the other hand.

That was what I was thinking. It would be just adding a few lines to [this test](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_core/robot_state/test/test_kinematic.cpp)